dev_tools/tests/test_nxdl_utils.py

```diff
@@ -136,3 +137,36 @@ def test_get_inherited_nodes():
         nx_name="NXiv_temp",
     )
     assert len(elist) == 4
+
+
+@pytest.mark.parametrize(
+    "hdf_name,concept_name,should_fit",
+    [
+        ("source_pump", "sourceType", False),
+        ("source_pump", "sourceTYPE", True),
+        ("source pump", "sourceTYPE", True),
+        ("source", "sourceTYPE", False),
```

Comment:
This is intended, right? So I always have to replace the upperstring part by something, i.e., not the empty string ""?

Reply:
Currently, that's the way how it is implemented. I did not particularly design it like this and I'm not sure what the old (buggy) NIAC version did. So for me it's up for discussion whether we want to allow the empty string, too.

Reply:
As discussed today we don't allow the empty string according to this [document](https://fairmat-nfdi.github.io/nexus_definitions/datarules.html)